src/binder/bind/read/bind_load_from.cpp

```diff
@@ -16,38 +16,49 @@ using namespace kuzu::catalog;
 namespace kuzu {
 namespace binder {
 
+static TableFunction getObjectScanFunc(const ObjectScanSource* objectSource,
+    main::ClientContext* clientContext) {
+    // Bind external database table
+    auto dbName = objectSource->objectNames[0];
+    auto attachedDB = clientContext->getDatabaseManager()->getAttachedDatabase(dbName);
+    if (attachedDB == nullptr) {
+        throw BinderException{stringFormat("No database named {} has been attached.", dbName)};
+    }
+    auto tableName = objectSource->objectNames[1];
+    auto attachedCatalog = attachedDB->getCatalogContent();
+    auto tableID = attachedCatalog->getTableID(tableName);
+    auto entry = attachedCatalog->getTableCatalogEntry(tableID);
+    auto tableEntry = ku_dynamic_cast<CatalogEntry*, TableCatalogEntry*>(entry);
+    return tableEntry->getScanFunction();
+}
+
 std::unique_ptr<BoundReadingClause> Binder::bindLoadFrom(const ReadingClause& readingClause) {
     auto& loadFrom = readingClause.constCast<LoadFrom>();
     TableFunction scanFunction;
     std::unique_ptr<TableFuncBindData> bindData;
     auto source = loadFrom.getSource();
     switch (source->type) {
     case ScanSourceType::OBJECT: {
-        auto objectSource = source->constPtrCast<ObjectScanSource>();
+        auto objectSource = reinterpret_cast<ObjectScanSource*>(source);
```

Comment:
I assume you are changing because you need to modify objectSource so const ptr no longer works.

In such case, either copy and modify or use a non-const ptrCast.